### PR TITLE
chore(vsc): upgrade @azure/identity and @azure/msal-node

### PR DESCRIPTION
**Pull request by tecton**
Upgrade dependencies to fix security issue in jsonwebtoken@8.5.1. ([Related issue](https://dev.azure.com/mseng/VSIoT/_componentGovernance/721/alert/101127?typeId=158182))
Referencing [this migration guide](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/migration-v1-v2.md) to upgrade @azure/identiy from v1 to v3.

### CODE REVIEW

File: packages/fx-core/package.json [hunk 1]
```diff
@@ -61,8 +61,8 @@
     "@azure/arm-sql": "^9.0.0",
     "@azure/arm-storage": "^17.2.1",
     "@azure/arm-subscriptions": "^5.0.0",
-    "@azure/identity": "^1.5.2",
-    "@azure/msal-node": "^1.0.0",
+    "@azure/identity": "^3.1.3",
```
Comment (swatDong) [inline packages/fx-core/package.json:64]: @xiaolang124 any specific test case needs to be added for such major version change?

Comment (xiaolang124) [reply packages/fx-core/package.json:64]: I think if e2e test cases work fine, this change is fine.

Comment (tecton) [reply packages/fx-core/package.json:64]: @xiaolang124 I've triggered the E2E test pipeline: https://github.com/OfficeDev/TeamsFx/actions/runs/4050434854
There are still several failures, it seems that they are not related to this change.

**Review by xiaolang124 — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

File: packages/fx-core/package.json (hunk 1 above)
Comment (tecton) [reply packages/fx-core/package.json]: Latest E2E: https://github.com/OfficeDev/TeamsFx/actions/runs/4060696750

**Pull request merged**